app/bundles/CampaignBundle/Model/EventLogModel.php

```diff
@@ -32,14 +36,15 @@ public function __construct(
         $this->eventScheduler = $eventScheduler;
     }
 
-    /**
-     * {@inheritdoc}
-     *
-     * @return \Mautic\CampaignBundle\Entity\LeadEventLogRepository
-     */
-    public function getRepository()
+    public function getRepository(): LeadEventLogRepository
     {
-        return $this->em->getRepository('MauticCampaignBundle:LeadEventLog');
+        $result = $this->em->getRepository(LeadEventLog::class);
+
+        if (!$result instanceof LeadEventLogRepository) {
+            throw new \RuntimeException('Wrong repository given.');
+        }
```

Comment:
Checks in this PR like this one could be also replaced by 
```suggestion
        \assert($result instanceof LeadEventLogRepository);
```

Reply:
Was thinking about assert, but this change also checks at runtime. Due to the low coverage we can spot these at runtime. But i can change all of these to assert no prob. WDYT?

Reply:
I'd recommend to use `/assert()` as it will fail on runtime in the dev and test environments. And it's quite hard (read waste of time) to write tests that will cover the code in these ifs. I'd suggest to replace it. Advantages:
- Shorter, readable code
- Works for IDEs and PHPSTAN
- Fails on test and dev environments
- Better performance in production

Reply:
See last commits